e2e/pages/Table/TableModule.ts

```diff
@@ -469,21 +378,46 @@ class TableModule {
     }
   }
 
-  async selectNonVisaFspPA() {
+  async selectFspPaPii({ shouldSelectVisa }: { shouldSelectVisa: boolean }) {
     await this.page.waitForSelector(TableModule.getCellValueTableRight(1, 4));
 
     const count = await this.paCell.count();
     for (let i = 1; i <= count; i++) {
-      const fsp = this.page.locator(TableModule.getCellValueTableRight(i, 4));
+      const fsp = this.page.locator(TableModule.getRow(i));
       const fspText = (await fsp.textContent())?.trim();
-      if (fspText !== visaFspIntersolve.displayName.en) {
+      const isVisaFsp = fspText?.includes(visaFspIntersolve.displayName.en);
+
+      if (
+        (shouldSelectVisa && isVisaFsp) ||
+        (!shouldSelectVisa && !isVisaFsp)
+      ) {
         await this.openPaPersonalInformation({ buttonIndex: i - 1 });
         return i;
       }
     }
     return 400;
   }
 
+  async openFspProfile({ shouldIncludeVisa }: { shouldIncludeVisa: boolean }) {
+    await this.page.waitForSelector(TableModule.getCellValueTableRight(1, 4));
+
+    const count = await this.paCell.count();
+    for (let i = 1; i <= count; i++) {
+      const fsp = this.page.locator(TableModule.getRow(i));
+      const fspText = (await fsp.textContent())?.trim();
+      const isVisaFsp = fspText?.includes(visaFspIntersolve.displayName.en);
+
+      if (
+        (shouldIncludeVisa && isVisaFsp) ||
+        (!shouldIncludeVisa && !isVisaFsp)
+      ) {
+        await this.clickOnPaNumber(i);
+        return i;
+      }
+    }
+    return 400;
```

Comment:
```suggestion
    return -1;
```

Same as in the other PR :)